onnxruntime/core/framework/session_state.cc

```diff
@@ -1098,12 +1098,55 @@ Status SessionState::FinalizeSessionState(const std::basic_string<PATH_CHAR_TYPE
                                   remove_initializers, constant_initializers_use_count);
 }
 
+static OrtValueIndex Index(const OrtValueNameIdxMap& ort_value_name_idx_map,
+                           const OrtValueName& name) {
+  OrtValueIndex result;
+  auto status = ort_value_name_idx_map.GetIdx(name, result);
+  ORT_THROW_IF_ERROR(status);
+  return result;
```

Comment:
Can we just return the status here and set the index as an out arg? 

Having a throw that we catch in OuterScopeNodeArgLocationAccumulator and convert back to a Status seems like an unnecessary binary size cost.

Reply:
Makes sense. Fixed.